crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -132,6 +178,61 @@ pub fn calculate_bounds(
     }
 }
 
+/// Updates [`Aabb`]s for [`Entities`](Entity) with [`Mesh`]es. This includes `Entities` that have
+/// been assigned new `Mesh`es as well as `Entities` whose `Mesh` has been directly mutated.
+///
+/// To opt out of bound calculation for an `Entity`, give it the [`NoFrustumCulling`] component.
+pub fn update_bounds(
+    mut commands: Commands,
+    meshes: Res<Assets<Mesh>>,
+    mut mesh_reassigned: Query<
+        (Entity, &Handle<Mesh>, &mut Aabb),
+        (Changed<Handle<Mesh>>, Without<NoFrustumCulling>),
+    >,
+    mut entity_mesh_rel: ResMut<EntityMeshRelationships>,
+    mut mesh_events: EventReader<AssetEvent<Mesh>>,
+    entities_lost_mesh: RemovedComponents<Handle<Mesh>>,
+) {
+    // If an entity's mesh handle was removed, unlink them.
+    for entity in entities_lost_mesh.iter() {
+        entity_mesh_rel.deregister(entity);
+    }
+
+    // If an entity with an Aabb was assigned a new mesh, move it to the new list and update its
+    // Aabb.
+    for (entity, mesh_handle, mut aabb) in mesh_reassigned.iter_mut() {
+        // Remove it from the previous list
+        entity_mesh_rel.deregister(entity);
+
+        // Add it to the new list.
+        entity_mesh_rel.register(entity, mesh_handle);
+
+        // Update its Aabb
+        if let Some(mesh) = meshes.get(mesh_handle) {
+            if let Some(new_aabb) = mesh.compute_aabb() {
+                *aabb = new_aabb;
```

Comment:
So would be a good idea to move the `register` and `deregister` inside the `if let` block here as well.

Reply:
Updated in b02be5c18b8f29a6ac8aac6bda3dc466dffed11a. Note that we _always_ deregister the entity<->mesh. I think that makes sense.